tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -3470,19 +3446,16 @@ SpirvVariable *DeclResultIdMapper::createSpirvStageVar(
     return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FullyCoveredEXT,
                                          isPrecise, srcLoc);
   }
-  // According to DXIL spec, the ShadingRate SV can only be used by GSOut,
-  // VSOut, or PSIn. According to Vulkan spec, the FragSizeEXT BuiltIn can only
-  // be used as PSIn.
   case hlsl::Semantic::Kind::ShadingRate: {
     switch (sigPointKind) {
```

Comment:
It seems `SV_ShadingRate` is only allowed to be used for `uint` type.
(See https://github.com/KhronosGroup/SPIRV-Tools/blob/master/test/val/val_builtins_test.cpp#L3842)

I asked this question to doublecheck #3676 .

Why don't we add type check here?

e.g.,
```
case hlsl::Semantic::Kind::ShadingRate: {
QualType ty;
if (!isScalarType(type, &ty) || !ty->isIntegerType()) {
// report error
}
```

`bool isScalarType(QualType type, QualType *scalarType)` is defined in tools/clang/lib/SPIRV/AstTypeProbe.cpp file.

Reply:
There will be an implicit float->uint type conversion in such case. Anyway, adding a type check won't hurt.